File: backend/src/test/java/org/luxons/sevenwonders/test/api/SevenWondersClient.java
package org.luxons.sevenwonders.test.api;

import java.util.concurrent.ExecutionException;
import java.util.concurrent.TimeoutException;

import org.hildan.jackstomp.JackstompClient;
import org.hildan.jackstomp.JackstompSession;

public class SevenWondersClient {

    private static final String WEBSOCKET_ENDPOINT = "/seven-wonders-websocket";

    private final JackstompClient client;

    public SevenWondersClient() {
        this(new JackstompClient());
    }

    public SevenWondersClient(JackstompClient client) {
        this.client = client;
    }

    public SevenWondersSession connect(String serverUrl)
            throws InterruptedException, ExecutionException, TimeoutException {
        JackstompSession session = client.connect(serverUrl + WEBSOCKET_ENDPOINT);
        return new SevenWondersSession(session);
    }

    public void stop() {
        client.stop();
    }
}

File: backend/src/test/kotlin/org/luxons/sevenwonders/test/api/SevenWondersClient.kt
package org.luxons.sevenwonders.test.api

import java.util.concurrent.ExecutionException
import java.util.concurrent.TimeoutException

import org.hildan.jackstomp.JackstompClient
import org.hildan.jackstomp.JackstompSession

class SevenWondersClient @JvmOverloads constructor(private val client: JackstompClient = JackstompClient()) {

    @Throws(InterruptedException::class, ExecutionException::class, TimeoutException::class)
    fun connect(serverUrl: String): SevenWondersSession {
        val session = client.connect(serverUrl + WEBSOCKET_ENDPOINT)
        return SevenWondersSession(session)
    }

    fun stop() {
        client.stop()
    }

    companion object {

        private val WEBSOCKET_ENDPOINT = "/seven-wonders-websocket"
    }
}
